piqe_ocp_lib/api/resources/ocp_nodes.py

```diff
@@ -257,3 +259,94 @@ def execute_command_on_a_node(self, node_name, command_to_execute):
             "Command - %s - execution status:\n" "RETCODE: %s\nSTDOUT: %s\nSTDERR: %s\n", command, ret, out, err
         )
         return ret, out, err
+
+    def get_master_nodes(self) -> Optional[ResourceList]:
+        """
+        Method that returns a list of master node objects
+        :param label_selector: Used to return a a list of nodes based on the provided label(s)
+        :return: V1NodeList on success. None on failure.
+        """
+        master_node_object_list = None
+        try:
+            master_node_object_list = self.get_all_nodes(label_selector="node-role.kubernetes.io/master")
+        except ApiException as e:
+            logger.error("Exception when calling method list_node: %s\n", e)
+        return master_node_object_list
+
+    def get_worker_nodes(self) -> Optional[ResourceList]:
+        """
+        Method that returns a list of worker node objects
+        :param label_selector: Used to return a a list of nodes based on the provided label(s)
+        :return: V1NodeList on success. None on failure.
+        """
+        worker_node_object_list = None
+        try:
+            worker_node_object_list = self.get_all_nodes(label_selector="node-role.kubernetes.io/worker")
+        except ApiException as e:
+            logger.error("Exception when calling method list_node: %s\n", e)
+        return worker_node_object_list
+
+    def is_node_schedulable(self, node_name: str) -> bool:
+        """
+        Return the node schedulable status
+        :param node_name:
+        :return: Node schedulable status, Either True or False. None on failure.
+        """
+        node_object = None
+        try:
+            node_object = self.ocp_nodes.get(name=node_name)
+            if node_object.spec.unschedulable:
+                logger.info("Node %s status is unschedulable" % node_name)
+                return False
+            else:
+                logger.info("Node %s status is schedulable" % node_name)
+                return True
+        except ApiException as e:
+            logger.error("Exception encountered while determining the node schedulable status: %s\n", e)
+        return node_object
+
+    def mark_node_schedulable(self, node_name: str) -> Optional[ResourceInstance]:
+        """
+        Return node which was marked as schedulable
+        :param node_name:
+        :return: A V1Node object. None on failure.
+        """
+        api_response = None
+        body = {
+            "spec": {
+                "unschedulable": False
+            }
+        }
+        schedulable_status = self.is_node_schedulable(node_name)
+        if schedulable_status:
```

Comment:
Docstring says this  method return an Object on success and None on failure. If node is already schedulable, this method will return None. I think we should either change the method signature so that the return type is a bool or perform a 'get' on the node when already schedulable so we can be consistent. I am leaning towards sticking to returning the node resource for the sake of consistency.